### ISSUE TITLE
Man help page shows up for objects even if `-- --help` isn't passed

### ISSUE THREAD

**Issue by plynch-chwy** [google/python-fire#213]
Based on my understanding of the [`_PrintResult`](https://github.com/google/python-fire/blob/master/fire/core.py#L241) function, it seems like the help page with possible methods will pop up if returning an object, even if `-- --help` isn't passed. It seems like the help page should only pop up if explicitly added.

Here is an example I came across:
I set up a fire script to connect to a sql database and run a sql file or code and return the connections cursor. Sometimes I want want to run a file creating a table or insert data without calling another method. Other times I may want to call `- fetchone` to see one row of results from a query. I don't want man pages to pop up the times where I don't call another method on the cursor, but that is what happens.

**Comment by plynch-chwy** [comment]
Was able to create a hacky work around by monkey patching the class `__str__` method:
```
vertica_python.vertica.cursor.Cursor.__str__ = lambda self: ""
```

**Comment by dbieber** [comment]
Yes, if the object returned is simple enough, then Fire will serialize it.
Similarly, if an object defines a custom \_\_str\_\_ method then Fire will serialize it.
If it's more complex and lacks a custom \_\_str\_\_ method, Fire will show the help screen.

I think this usually gives good behavior, though admittedly in the example you give you get a better experience using the hacky workaround.

We've talked about enabling more direct configuration of how outputs are formatted (so you wouldn't need to use such a hacky workaround), but it's not on the short term roadmap yet.
